openapi/data_repository_service.swagger.yaml

```diff
@@ -194,57 +181,84 @@ definitions:
           sha512
   Bundle:
     type: object
-    required: ['id', 'object_ids', 'created', 'updated', 'version', 'checksums']
     properties:
       id:
         type: string
-        description: |-
+        description: >-
           An identifier, unique to this Data Bundle
-      object_ids:
-        type: array
-        items:
-          type: string
+      name:
+        type: string
         description: |-
-          The list of Data Objects that this Data Bundle contains.
+          A string that can be optionally used to name a Data Bundle.
+      size:
+        type: string
+        format: int64
+        description: |-
+          The cumulative size of all Data Objects and Bundles (within contents) in bytes.
       created:
         type: string
         format: date-time
-        description: |-
-          Timestamp of object creation in RFC3339.
+        description: >-
+          Timestamp of Bundle creation in RFC3339.
       updated:
         type: string
         format: date-time
-        description: |-
-          Timestamp of update in RFC3339, identical to create timestamp in systems
-          that do not support updates.
+        description: >-
+          Timestamp of Bundle update in RFC3339, identical to create timestamp in
+          systems that do not support updates.
       version:
         type: string
-        description: |-
-          A string representing a version, some systems may use checksum, a RFC3339
-          timestamp, or incrementing version number. For systems that do not support
-          versioning please use your update timestamp as your version.
+        description: >-
+          A string representing a version, some systems may use checksum, a
+          RFC3339 timestamp, or incrementing version number.
+      mime-type:
+        type: string
+        description: >-
+          A string providing the mime-type of the Data Bundle.
+        example: 'application/json'
       checksums:
         type: array
+        description: >-
+          At least one checksum must be provided.
+          The Data Bundle checksum is computed over a sorted concatenation of all the 
+          checksums within the top-level 'contents' of the Bundle (not recursive). The list of Data
+          Object checksums are sorted alphabetically (hex-code) before concatenation and a further
```

Comment:
As written, the checksum only includes the checksums of the top-level Objects -- is that right? I would have expected it to also include the checksums of any top-level Bundles. Maybe just replace "Data Object checksums" with "Object or Bundle checksums"?

Reply:
Tx. Description updated.